eng/scripts/Language-Settings.ps1

```diff
@@ -152,4 +153,18 @@ function Publish-java-GithubIODocs ($DocLocation, $PublicArtifactLocation)
       }
     }
   }
+}
+
+function Generate-java-GithubIODocIndex () {
+  $metadata = Get-java-Metadata 
```

Comment:
```suggestion
  $metadata = Get-java-Metadata | Where-Object { $_.Hide -ne 'true' -and $_.GroupId -eq 'com.azure' }
```

I would do the filtering right here instead of passing in the filtering options.

Reply:
Agreed for most of the cases, but we have the case where the artifact only has the group id: "com.microsoft.azure". That's why I checked the logic here.
```
function Get-PkgMetadata([Object[]]$metadata, [String]$artifactName, [Hashtable]$filterOptions) {
    $filteredPkg = $metadata | ? { $_.Package -eq $artifactName }
    if (!$filteredPkg) {
        return
    }
    # No filter logic if only one artifact returned
    if ($filteredPkg.length -eq 1) {
        return $filteredPkg[0]
    }
    foreach ($filter in $filterOptions.getEnumerator()) {
        $filterKey = $filter.Key.Trim()
        $filterValue = $filter.Value.Trim()
        $filteredPkg = $filteredPkg | ? { $_.$filterKey -eq $filterValue }
    }
    if ($filteredPkg.length -ne 1) {
        LogWarning "We fetch out $($filteredPkg.length) artifact(s) for the same package name $artifactName. "
        LogWarning "Please check csv of Azure/azure-sdk/_data/release/latest repo if this is intended. "
        return 
    }
    return $filteredPkg[0]
}
```

Reply:
Ok if there are cases like that then I'd do the filtering and only remove duplicates if the package names exists. Something like:

```
$metadata = Get-Metadata | Where-Object { $_.Hide -ne 'true' }
$clientPackages = $metadata | Where-Object { $_.GroupId -eq 'com.azure' } | Foreach-Object { $_.Package }
$metadata = $metadata | Where-Object { $_.GroupId -eq 'com.azure' -or !$clientPackages.Contains($_.Package) } 
```